### DISCUSSION TITLE
OpenBSD SSL verification

### DISCUSSION THREAD

**Discussion by caywin25** [FreshRSS/FreshRSS#6288] [Q&A] [answered]
Hello

I just installed freshrss.
Thanks for your work, the software is great.

However, I find myself facing a problem.
I have to disable SSL verification on all my feeds otherwise they won't update.

In my logs I have this error:
cURL error 77: error setting certificate verify locations: CAfile: /etc/ssl/cert.pem CApath: none

Has this problem ever happened to anyone?
Thank you for your help.

**Comment by Frenzie** [comment]
How are you running it exactly? It sounds like you're missing some fairly vital system package.

**Comment by caywin25** [comment]
Thank you for your reply.
The only vital package for my problem is curl or php-curl.
Both are well installed :(
Looks like it's a path problem: CApath: none

**Reply by Frenzie** [reply]
Okay, but which distro?

**Reply by caywin25** [reply]
Openbsd :3

**Reply by Frenzie** [reply]
If you can find where the certificate is actually located you could probably fix it in php.ini (something like `curl.cainfo`), but normally I wouldn't expect to have to do anything there unless you fully installed it from scratch. Maybe someone else knows more about OpenBSD.

**Reply by Alkarex** [reply]
I do not know with OpenBSD, but other distros have a package and a command for CA certs. E.g. for Debian / Ubuntu:
* https://packages.debian.org/sid/ca-certificates

Does a cURL with an HTTPS Web site work for you?

```sh
curl -v https://github.com
```

You could alternatively consider running FreshRSS [from Docker](https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md), which does not need to setup those things.

**Reply by caywin25** [reply]
Hello @Alkarex 

The ca-certificate equivalent for Openbsd is cert.pem (if I'm not saying something stupid)

Regarding curl -v https://github.com yes it works:

```
$ curl -v https://github.com
* Host github.com:443 was resolved.
* IPv6: (none)
* IPv4: 140.82.121.4
*   Trying 140.82.121.4:443...
* Connected to github.com (140.82.121.4) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=github.com
*  start date: Mar  7 00:00:00 2024 GMT
*  expire date: Mar  7 23:59:59 2025 GMT
* [...]
```

If I try to check the freshrss updates in the logs I get this error:

```
Error during update (HTTP code 0): error setting certificate verify locations: CAfile: /etc/ssl/cert.pem CApath: none
```

Which shows that php doesn't seem to know where the certificates are located (I think).

I will try to find the solution.

With openbsd we use libressl and not openssl.
Wouldn't the problem come from there?
In the meantime, is it "dangerous" not to do this SSL verification with my RSS feeds?

Thanks

**Reply by Frenzie** [reply]
> In the meantime, is it "dangerous" not to do this SSL verification with my RSS feeds?

Not really, give or take obvious things like passwords if applicable, and of course which URL you're requesting. There's the potential but presumably negligible risk that someone could intercept the request and insert some nonsense.

> Which shows that php doesn't seem to know where the certificates are located (I think).

`/etc/ssl/cert.pem` seems to be the same path as in your command line curl command. My guess would be (one of) the certificates is considered invalid. Could you try the curl command with the same URL as the one that fails?

**Reply by caywin25** [reply, marked as answer]
I finally found the solution
You had to copy /etc/ssl/cert.pem into the web server chroot.

Everything is good and works perfectly :)
Thank you both for your leads and help.

Have a good day :)
